models/image.py

```diff
@@ -877,72 +702,110 @@ def from_images(cls, images):
         output.exp_time = sum([image.exp_time for image in images])
 
         # start MJD and end MJD
-        output.mjd = min([image.mjd for image in images])
-        output.end_mjd = max([image.end_mjd for image in images])
+        output.mjd = images[0].mjd  # assume sorted by start of exposures
+        output.end_mjd = max([image.end_mjd for image in images])  # exposure ends are not necessarily sorted
 
         # TODO: what about the header? should we combine them somehow?
         output.header = images[0].header
         output.raw_header = images[0].raw_header
 
-        output.source_images = images
         base_type = images[0].type
         if not base_type.startswith('Com'):
             output.type = 'Com' + base_type
 
+        output.upstream_images = images
+
+        # mark the first image as the reference (this can be overriden later when actually warping into one image)
+        output.ref_image_index = 0
```

Comment:
(Again associated with comment above about using image id rather than index to identify the reference.)